docs/source/reference/optimizer.md

```diff
@@ -0,0 +1,408 @@
+<!--
+SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+SPDX-License-Identifier: Apache-2.0
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+# NVIDIA NeMo Agent toolkit Optimizer Guide
+
+Welcome to the NeMo Agent toolkit Optimizer guide. This document provides a comprehensive overview of how to use the NeMo Agent toolkit Optimizer to tune your NeMo Agent toolkit workflows.
+
+## Introduction
+
+The NeMo Agent toolkit Optimizer is a powerful tool for automated hyperparameter tuning and prompt engineering for your NeMo Agent toolkit workflows. It allows you to define a search space for your workflow's parameters and then intelligently searches for the best combination of parameters based on the evaluation metrics you specify.
+
+This guide will walk you through:
+- Configuring the optimizer.
+- Making your workflow `parameters.optimizable`.
+- Running the optimizer from the command line.
+- Understanding the output of the optimizer.
+
+## How it Works
+
+The NeMo Agent toolkit Optimizer uses a combination of techniques to find the best parameters for your workflow:
+
+- Numerical hyperparameter optimization uses [Optuna](https://optuna.org/).
+- Prompt optimization uses a genetic algorithm (GA) that evolves a population of prompt candidates over multiple generations using LLM-powered mutation and optional recombination.
+
+```
```

Comment:
The alignment of some entities in the chart seem off. Do we support mermaid diagrams with documentation? If so, this could be a good candidate